### ISSUE TITLE
[BUG] GitHub + macOS + Node 14: All tests pass but process exits with code 134

### ISSUE THREAD

**Issue by jhildenbiddle** [microsoft/playwright#3090]
**Context:**
- Playwright Version: 1.2.0
- Operating System: macOS
- Node version: 14
- Browser: Webkit
- Extra: Running as GitHub Action using [microsoft/playwright-github-action](https://github.com/microsoft/playwright-github-action) v1.3.0

```
Running one project: e2e
PASS browser: chromium e2e tests/e2e/example.test.js
PASS browser: firefox e2e tests/e2e/example.test.js
PASS browser: webkit e2e tests/e2e/example.test.js
PASS browser: chromium e2e tests/e2e/vue.test.js
PASS browser: firefox e2e tests/e2e/vue.test.js
PASS browser: webkit e2e tests/e2e/vue.test.js

Test Suites: 6 passed, 6 of 2 total
Tests:       24 passed, 24 total
Snapshots:   0 total
Time:        21.526 s
Ran all test suites.
Assertion failed: (0), function uv_close, file ../deps/uv/src/unix/core.c, line 178.
/Users/runner/work/_temp/241fae1a-1600-440c-9c4a-dc01208caa50.sh: line 1:  3699 Abort trap: 6
npm run test:jest-e2e -- -ci
##[error]Process completed with exit code 134.
```

Tests pass with all other os/node versions:

* macOS + Node 10/12 (not 14)
* Ubuntu + Node 10/12/14
* Windows + Node 10/12/14

**Comment by jhildenbiddle** [comment]
Just received the same error using the setup above but with macOS + Node 12.18.2. I've seen playwright test on macOS + Node 12 on GitHub though, so perhaps GitHub's macOS setup is just flakey.

**Comment by aslushnikov** [comment]
@jhildenbiddle This looks like some `fsevents` or `chokidar` error. There are some claims that upgrading to `chokidar` v3 fixes this error. do you actually use `chokidar`? If yes, what version?

**Comment by jhildenbiddle** [comment]
Thanks for the reply, @aslushnikov.

We have `chokidar` as a dev dependency (currently v3.2.1). It's only used as part of our rollup build, so I'd be surprised if that's what causing the issue with our e2e tests on macOS.

Best I can tell from the usual searches is that exit code 134 is likely related to running out of memory. I've tried using both [`--runInBand` for Jest](https://jestjs.io/docs/en/cli#--runinband) and the [`runInProcess` for jest-image-snapshot](https://github.com/americanexpress/jest-image-snapshot#%EF%B8%8F-api) hoping one of those would address the issue (even if tests run slower, it's a start) but... no luck.

I'm guessing this issue is better directed toward the GitHub team. If you have any ideas I'm all ears, but there will be no hard feelings if the response from the Playwright team is "not our issue".

**Comment by yury-s** [comment]
I doubt it is cause by OOMing, the problem sounds very similar to [this node issue](https://github.com/nodejs/node/issues/32389) and [this webpack one](https://github.com/webpack/webpack/issues/10037). I wonder if the received SIGABRT has anything to do with Playwright or it's a test runner problem.

@jhildenbiddle does it reproduce with other browsers or just WebKit?

**Comment by jhildenbiddle** [comment]
@yury-s --

Turns out that setting [`runInProcess`](https://github.com/americanexpress/jest-image-snapshot#%EF%B8%8F-api) to `true` for [jest-image-snapshot](https://github.com/americanexpress/jest-image-snapshot) solves the issue so I recreated an issue there: https://github.com/americanexpress/jest-image-snapshot/issues/231.

For clarification, I believe the issue is macOS- and/or GitHub-specific, not Webkit-specific. All tests pass without issue locally on macOS and on both Ubuntu and Windows using GitHub Actions.

**Comment by jhildenbiddle** [comment]
Thanks to @aslushnikov and @yury-s I was able to track down the issue. The linked issues provided by @yury-s convinced me that an outdated version of chokidar was likely the cause. I ran the following CLI command to determine if a dependency on an outdated version existed in our project:

```bash
npm ls chokidar
```

Sure enough, this was the result:

```bash
├── chokidar@3.4.1 
├─┬ live-server@1.2.1
  └── chokidar@2.1.8
```

Turns out [live-server](https://github.com/tapio/live-server) is still using chokidar v2. A live-server [PR has been created](https://github.com/tapio/live-server/pull/343) to address the issue, but since that PR is now four months old with no sign of activity, we made the switch to [browser-sync](https://www.browsersync.io).  Very happy to report that all tests pass without issue now.

Thanks again for your help. The linked issues were critical in identifying the problem. Very much appreciated!
